06.bowling_object/frame.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  def initialize(index, first_shot, second_shot = nil, third_shot = nil)
+    @index = index
+    @shots = [first_shot, second_shot, third_shot].compact
+  end
+
+  def index
+    @index
+  end
+
+  def calc_total_score(frames)
+    next_frame = frames[index + 1]
+    after_next_frame = frames[index + 2]
+    calc_score + calc_bonus_score(next_frame, after_next_frame)
+  end
+
+  protected
+
+  def calc_score(shot_count = @shots.size)
+    @shots.take(shot_count).sum(&:score)
+  end
+
+  def strike?
+    @shots.first.strike?
+  end
+
+  private
+
+  def spare?
+    calc_score == 10 && !strike?
+  end
+
+  def calc_bonus_score(next_frame, after_next_frame)
+    return 0 unless next_frame
```

Comment:
```suggestion
    return 0 if last_frame?
```

最後のフレームだとボーナス計算はなしですよ、みたいに読めるようにするのも一手です。（好みの問題）

Reply:
`last_frame?`メソッドを追加しました。
意図がわかりやすくなりました。